src/components/utils/rqb-inputs/custom-react-query-builder.tsx

```diff
@@ -0,0 +1,95 @@
+/**
+ * Copyright (c) 2023, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import Grid from '@mui/material/Grid';
+import { QueryBuilderMaterial } from '@react-querybuilder/material';
+import {
+    Field,
+    formatQuery,
+    OptionList,
+    QueryBuilder,
+    RuleGroupTypeAny,
+} from 'react-querybuilder';
+import {
+    getOperators,
+    queryValidator,
+} from '../../dialogs/filter/expert/expert-filter-utils';
+import { ErrorInput, FieldErrorAlert } from '@gridsuite/commons-ui';
+import { useIntl } from 'react-intl';
+import { combinatorType } from '../../dialogs/filter/expert/expert-filter-constants';
+import { useFormContext } from 'react-hook-form';
+import RemoveButton from 'components/utils/rqb-inputs/remove-button';
+import CombinatorSelector from 'components/utils/rqb-inputs/combinator-selector';
+import AddButton from 'components/utils/rqb-inputs/add-button';
+import ValueEditor from './value-editor';
+import { EXPERT_FILTER_QUERY } from '../../dialogs/filter/expert/expert-filter-form';
+
+interface ExpertFilterRQBProps {
```

Comment:
I think using ComponentName+Props is a better naming convention, i.e. CustomReactQueryBuilderProps